### PR TITLE
WS Reconnect changes

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [ ] documentation updated

**Squash commit message**

* Add reconnect errors to current state if existing, instead of separate error log
* Only allow one reconnect function on websocket reconnect.
* Rewrite connect/re-connect to look at session_created, session_attached pushed messages

**Info**

Optional informative text (not to be included in commit message) relevant to reviewers.

### CODE REVIEW

**Review by foh79 — APPROVED** [review]
Looks good to me but Anton should take a look

File: wsdialer/wsdialer.go [hunk 1]
```diff
@@ -181,7 +181,18 @@ func (dialer *WsDialer) ReadMessage() (int, []byte, error) {
 		data = append(data, m.Payload...)
 	}
 
-	if err == io.EOF {
+	disconnected := false
+	switch err := err.(type) {
+	// We might want to check sub errors, but have seen both net.ErrClosed and os.SyscallError on disconnect to handling all OpErrors as disconnects for now
```
Comment (atluq) [inline wsdialer/wsdialer.go:186]: 👌

**Review by atluq — APPROVED** [review]
Great work!

Consider making a type `ConnectFunc` out of `func(bool) (string, error)` so it can be documented. I figured out what the bool and string was, but it would be great to have quick access to it when reading the code.

File: wsdialer/wsdialer.go (hunk 1 above)
Comment (DnlLrssn) [reply wsdialer/wsdialer.go:186]: ah true would makes sense will do that.

**Review by atluq — APPROVED** [review]
Great docs on the connect func. It might be appropriate to use it in some more places.

**Review by foh79 — APPROVED** [review]

**Review by atluq — APPROVED** [review]
Looks good!

**Pull request merged**